### DISCUSSION TITLE
Is it possible to have 2 default providers (one https and one ws)?

### DISCUSSION THREAD

**Discussion by samlaf** [NomicFoundation/hardhat#2390] [General]
Hello,

is it possible for hardhat to create 2 default providers for a given network?
I often need to subscribe to events using a websocket provider, and I need to create this provider from scratch.
I might be unaware of this feature, but I was thinking it would be nice if hardhat.config.ts accepted for example
```
polygon: {
   url: `https://mainnet.infura.io/v3/${process.env.WEB3_INFURA_PROJECT_ID}`,
   ws-url: `ws://mainnet.infura.io/ws/v3/${process.env.WEB3_INFURA_PROJECT_ID}` 
}
```
and made two providers available for example through `ethers.httpProvider` and `ethers.wsProvider` or something of that sort.

Is this already possible and I am just not aware of how to access this functionality?

**Comment by alcuadrado** [comment]
Hardhat only supports 1 main connection at the moment. This may change in the future, but that will imply a major change, so it won't happen in the short-term.
